### PR TITLE
Add more functions for experimental tensor

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
### Changes

Add function for experimental tensor 
- concatenate
- logical_or
- masked_mean
- masked_median
- median
- percentile

### CODE REVIEW

File: nncf/experimental/tensor/functions/numeric.py [hunk 1]
```diff
@@ -633,3 +662,86 @@ def diag(a: Tensor, k: int = 0) -> Tensor:
...
+    :param x1: The input tensor.
+    :param x2: The tensor to compute or with.
+    :return: Result of elementwise or operation between input_ and other tensor.
+    """
+    return Tensor(logical_or(x1.data, unwrap_tensor_data(x2)))
+
+
+@functools.singledispatch
+@tensor_guard
+def zero_elements(x: Tensor) -> Tensor:
```
Comment (alexsu52) [inline nncf/experimental/tensor/functions/numeric.py:683]: Could you share user scenario or links to torch or numpy functions?

File: nncf/experimental/tensor/functions/torch_numeric.py [hunk 2]
```diff
@@ -299,3 +330,41 @@ def _(a: torch.Tensor, axis: int = -1, descending=False, stable=False) -> torch.
...
+
+
+@numeric.masked_mean.register(torch.Tensor)
+def _(
+    x: torch.Tensor, mask: Optional[torch.Tensor], axis: Union[int, Tuple[int, ...], List[int]], keepdims=False
+) -> torch.Tensor:
+    if mask is None:
+        return torch.mean(x, axis=axis, keepdims=keepdims)
+    device = x.device
+    masked_x = np.ma.array(x.detach().cpu().numpy(), mask=mask.detach().cpu().numpy())
```
Comment (alexsu52) [inline nncf/experimental/tensor/functions/torch_numeric.py:352]: It looks like not optimal implementation for torch backend in practice for gpu. Could you think about more optimal implementation without copying tensor in gpu case?

File: nncf/experimental/tensor/functions/torch_numeric.py [hunk 3]
```diff
@@ -299,3 +330,41 @@ def _(a: torch.Tensor, axis: int = -1, descending=False, stable=False) -> torch.
...
+
+
+@numeric.masked_median.register(torch.Tensor)
+def _(
+    x: torch.Tensor, mask: Optional[torch.Tensor], axis: Union[int, Tuple[int, ...], List[int]], keepdims=False
+) -> torch.Tensor:
+    if mask is None:
+        return torch.median(x, axis=axis, keepdims=keepdims)
+    device = x.device
+    masked_x = np.ma.array(x.detach().cpu().numpy(), mask=mask.detach().cpu().numpy())
```
Comment (alexsu52) [inline nncf/experimental/tensor/functions/torch_numeric.py:366]: It looks like not optimal implementation for torch backend in practice  for gpu. Could you think about more optimal implementation without copying tensor in gpu case?

File: nncf/experimental/tensor/functions/torch_numeric.py (hunk 2 above)
Comment (AlexanderDokuchaev) [reply nncf/experimental/tensor/functions/torch_numeric.py:355]: @daniil-lyakhov What limitations in torch led to the need to use numpy?

Comment (daniil-lyakhov) [reply nncf/experimental/tensor/functions/torch_numeric.py:352]: As `torch.masked_tensor` is an prototype (https://pytorch.org/docs/stable/masked.html) and does not implement neither median nor quantile, I decided to use numpy instead. As for now, it seems possible to use `torch.mean` with a torch masked tensor

**Review by daniil-lyakhov — APPROVED** [review]

File: nncf/experimental/tensor/functions/torch_numeric.py (hunk 2 above)
Comment (AlexanderDokuchaev) [reply nncf/experimental/tensor/functions/torch_numeric.py:352]: https://github.com/openvinotoolkit/nncf/blob/24f0ee707cd5a35abf351f8647efc0fbd0ee273b/nncf/torch/tensor_statistics/collectors.py#L90-L100

File: nncf/experimental/tensor/functions/torch_numeric.py (hunk 3 above)
Comment (AlexanderDokuchaev) [reply nncf/experimental/tensor/functions/torch_numeric.py:366]: https://github.com/openvinotoolkit/nncf/blob/24f0ee707cd5a35abf351f8647efc0fbd0ee273b/nncf/torch/tensor_statistics/collectors.py#L103-L114

File: nncf/experimental/tensor/functions/torch_numeric.py (hunk 2 above)
Comment (alexsu52) [reply nncf/experimental/tensor/functions/torch_numeric.py:352]: I'm not sure, but it looks like you can implement it something like that:
```
num = torch.sum(mask, axis)
y = torch.sum(x * mask, axis) / num
```

File: nncf/experimental/tensor/functions/numeric.py (hunk 1 above)
Comment (AlexanderDokuchaev) [reply nncf/experimental/tensor/functions/numeric.py]: ok, removed

File: nncf/experimental/tensor/functions/torch_numeric.py (hunk 2 above)
Comment (AlexanderDokuchaev) [reply nncf/experimental/tensor/functions/torch_numeric.py]: reworked by torch.nanmean

File: nncf/experimental/tensor/functions/torch_numeric.py (hunk 3 above)
Comment (AlexanderDokuchaev) [reply nncf/experimental/tensor/functions/torch_numeric.py]: reworked by torch.nanmedian

**Review by daniil-lyakhov — APPROVED** [review]
